### DISCUSSION TITLE
How to drag a window without the titlebar by a TextBox?

### DISCUSSION THREAD

**Discussion by MPWrobel** [picoe/Eto#1931] [Q&A]
I want to make a window without the titlebar (`WindowStyle="None"`) moveable. I know that there is `MovableByWindowBackground` property, but I want to fill the entire window with a TextBox, so that doesn't help. 

In Electron for [example](https://github.com/electron/electron/blob/master/docs/api/frameless-window.md#draggable-region) this can be achieved by setting `-webkit-app-region` property of a TextField to `drag`. Any ideas how to do this using Eto?

**Comment by cwensley** [comment]
There is no support for this in Eto.

One way would be to handle the MouseDown/MouseMove/etc to wire that up manually, but I imagine that will interfere with the operation of the TextBox.

Another way would be to figure out how to do this (if even possible) in each of the platforms you wish to target (Wpf, WinForms, Mac, Gtk, etc), and apply a native style to the control to enable it.

**Reply by MPWrobel** [reply]
I tried the MouseDown/MouseMove approach and the main issue was that MouseMove event isn't triggered when mouse button is down because the mouse is captured. If there was a way to release it prematurely, then I think this could work.

Otherwise I will maybe try to implement it on every platform independently. I'm pretty sure that it's possible.

Thanks for help!

**Reply by cwensley** [reply]
You need to set `e.Handled = true` during the `MouseDown` event to capture the mouse to get `MouseMove` to trigger.  Otherwise it will be handled by the TextBox itself, e.g. setting cursor position, etc.
